### PR TITLE
feat: angular 10 update

### PR DESCRIPTION
**Pull request by saithis**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,59 +22,58 @@
...
+    "@angular/platform-browser": "~10.0.5",
+    "@angular/platform-browser-dynamic": "~10.0.5",
+    "@angular/router": "~10.0.5",
+    "ngx-mat-select-search": "^3.0.1",
     "nouislider": "^14.0.2",
     "rxjs": "~6.5.5",
-    "tslib": "^1.13.0",
-    "zone.js": "~0.10.3"
+    "tslib": "^2.0.0",
+    "zone.js": "~0.10.2"
```
Comment (Yirica) [inline package.json:41]: zone.js  downgraded? 0.10.3 to 0.10.2)

File: projects/components/select/src/select.component.spec.ts [hunk 2]
```diff
@@ -1,451 +1,454 @@
...
+  const sortSelectTriggerEl = fixture.debugElement.nativeElement.querySelectorAll('.mat-select-trigger').item(0) as HTMLElement;
+  sortSelectTriggerEl.dispatchEvent(new MouseEvent('click'));
+  fixture.detectChanges();
+  await fixture.whenRenderingDone();
+}
+// async function getMatOptionsNodes(): Promise<NodeListOf<HTMLElement>> {
+//   const selectPanelNode = document.querySelector('.mat-select-panel');
+//   const matOptionNodes = selectPanelNode.querySelectorAll('mat-option') as NodeListOf<HTMLElement>;
+//   return matOptionNodes;
+// }
```
Comment (Yirica) [inline projects/components/select/src/select.component.spec.ts:440]: remove commented code?

File: projects/prosoft-components-demo/tsconfig.app.json [hunk 3]
```diff
@@ -1,13 +1,9 @@
 {
-  "extends": "../../tsconfig.json",
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "outDir": "../../out-tsc/app",
     "types": []
   },
-  "include": ["src/**/*.ts"],
-  "exclude": ["src/test.ts", "src/**/*.spec.ts"],
```
Comment (Yirica) [inline projects/prosoft-components-demo/tsconfig.app.json:8]: Not so sure whether this should be removed or not.

File: tsconfig.json [hunk 4]
```diff
@@ -1,11 +1,39 @@
+/*
+  This is a "Solution Style" tsconfig.json file, and is used by editors and TypeScript’s language server to improve development experience.
+  It is not intended to be used to perform a compilation.
+
+  To learn more about this file see: https://angular.io/config/solution-tsconfig.
+*/
+
+/*
+This doesn't ork with our path mappings. See: https://github.com/microsoft/TypeScript/issues/39632
```
Comment (Yirica) [inline tsconfig.json:9]: I'm pretty sure it should be either work or, in some unexpected cases, orc :D

File: package.json (hunk 1 above)
Comment (saithis) [reply package.json:41]: This results in the same version because of the ~

File: projects/prosoft-components-demo/tsconfig.app.json (hunk 3 above)
Comment (saithis) [reply projects/prosoft-components-demo/tsconfig.app.json:8]: The angular migration removed this

File: tsconfig.json (hunk 4 above)
Comment (saithis) [reply tsconfig.json:9]: :D

**Pull request merged**